<a href="https://colab.research.google.com/github/shabana1408/Project-2/blob/main/Project_2_Part_1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Project 2 - Part 1

## Requirements

Propose two possible datasets you would like to work with and indicate which is your first and which is your second choice.

Upload and shown the .head() of each of your data sets. For each of the proposed datasets, include the following information in text cells:

1. Source of data

2. Brief description of data

3. What is the target?

4. What does one row represent? (A person? A business? An event? A product?)

5. Is this a classification or regression problem?

6. How many features does the data have?

7. How many rows are in the dataset?

8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

## Imports

In [2]:
import pandas as pd

In [1]:
# mount google drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [10]:
pd.set_option('display.max_columns', None)

## First choice: Fetal Health Classification

Load the data

In [5]:
fpath1 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/fetal_health.csv'

df1 = pd.read_csv(fpath1)

Inspect the data

In [11]:
df1.head()

baseline value  accelerations  fetal_movement  uterine_contractions  \
0           120.0          0.000             0.0                 0.000   
1           132.0          0.006             0.0                 0.006   
2           133.0          0.003             0.0                 0.008   
3           134.0          0.003             0.0                 0.008   
4           132.0          0.007             0.0                 0.008   

   light_decelerations  severe_decelerations  prolongued_decelerations  \
0                0.000                   0.0                       0.0   
1                0.003                   0.0                       0.0   
2                0.003                   0.0                       0.0   
3                0.003                   0.0                       0.0   
4                0.000                   0.0                       0.0   

   abnormal_short_term_variability  mean_value_of_short_term_variability  \
0                             73.0                                   0.5   
1                             17.0                                   2.1   
2                             16.0                                   2.1   
3                             16.0                                   2.4   
4                             16.0                                   2.4   

   percentage_of_time_with_abnormal_long_term_variability  \
0                                               43.0        
1                                                0.0        
2                                                0.0        
3                                                0.0        
4                                                0.0        

   mean_value_of_long_term_variability  histogram_width  histogram_min  \
0                                  2.4             64.0           62.0   
1                                 10.4            130.0           68.0   
2                                 13.4            130.0           68.0   
3                                 23.0            117.0           53.0   
4                                 19.9            117.0           53.0   

   histogram_max  histogram_number_of_peaks  histogram_number_of_zeroes  \
0          126.0                        2.0                         0.0   
1          198.0                        6.0                         1.0   
2          198.0                        5.0                         1.0   
3          170.0                       11.0                         0.0   
4          170.0                        9.0                         0.0   

   histogram_mode  histogram_mean  histogram_median  histogram_variance  \
0           120.0           137.0             121.0                73.0   
1           141.0           136.0             140.0                12.0   
2           141.0           135.0             138.0                13.0   
3           137.0           134.0             137.0                13.0   
4           137.0           136.0             138.0                11.0   

   histogram_tendency  fetal_health  
0                 1.0           2.0  
1                 0.0           1.0  
2                 0.0           1.0  
3                 1.0           1.0  
4                 1.0           1.0

In [12]:
df1.shape
print(f'There are {df1.shape[0]} rows and {df1.shape[1]} columns.')
print(f'The rows represent {df1.shape[0]} observations, and the columns represent {df1.shape[1]-1} features and 1 target variable.')

There are 2126 rows and 22 columns.
The rows represent 2126 observations, and the columns represent 21 features and 1 target variable.


1. Source of the data

> https://www.kaggle.com/datasets/andrewmvd/fetal-health-classification

2. Brief description of data

> This dataset contains 2126 rows and 22 columns. Data is extracted from cardiotocogram exams, which were then classified by three expert obstetritians into 3 classes:
> - Normal
> - Suspect
> - Pathological

3. What is the target?

> `fetal_health` with values
> - 1 (Normal)
> - 2 (Suspect) and
> - 3 (Pathological)

4. What does one row represent?

> Each row represents the results from one cardiotocogram exam.

5. Is this a classification or regression problem?

> Classification

6. How many features does the data have?

> The dataset has 21 features.

7. How many rows are in the dataset?

> There are 2 126 rows in the dataset.


8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

> I don't foresee any challenges at this time.

## Second choice: Fitness Club Classification

Load the data

In [13]:
fpath2 = '/content/drive/MyDrive/CodingDojo/02-MachineLearning/Week08/Data/fitness_club.csv'

df2 = pd.read_csv(fpath2)

Inspect the data

In [14]:
df2.head()

booking_id  months_as_member  weight days_before day_of_week time  \
0           1                17   79.56           8         Wed   PM   
1           2                10   79.01           2         Mon   AM   
2           3                16   74.53          14         Sun   AM   
3           4                 5   86.12          10         Fri   AM   
4           5                15   69.29           8         Thu   AM   

   category  attended  
0  Strength         0  
1      HIIT         0  
2  Strength         0  
3   Cycling         0  
4      HIIT         0

In [15]:
df2.shape
print(f'There are {df2.shape[0]} rows and {df2.shape[1]} columns.')
print(f'The rows represent {df2.shape[0]} observations, and the columns represent {df2.shape[1]-1} features and 1 target variable.')

There are 1500 rows and 8 columns.
The rows represent 1500 observations, and the columns represent 7 features and 1 target variable.


1. Source of the data

> https://www.kaggle.com/datasets/ddosad/datacamps-data-science-associate-certification

2. Brief description of data

> This dataset contains 1 500 rows and 8 columns. The dataset represents fitness club bookings for GoalZone, a fitness club in Canada. GoalZone offers a range of fitness classes in two capacities - 25 members and 15 members. Some classes are always fully booked, noting that fully booked classes often have a low attendance rate.
>
> GoalZone wants to increase the number of spaces available for classes by predicting whether the member will attend the class or not.

3. What is the target?

> `Attended` with values
> - 0 (Not attended) and
> - 1 (Attended).

4. What does one row represent?

> Each row represents a booking for a fitness club session.

5. Is this a classification or regression problem?

> Classification

6. How many features does the data have?

> The dataset has 7 features.

7. How many rows are in the dataset?

> There are 1 500 rows in the dataset.


8. What, if any, challenges do you foresee in cleaning, exploring, or modeling this dataset?

> I don't foresee any challenges at this time.